### ISSUE TITLE
[device_info_plus] Get current device information as map

### ISSUE THREAD

**Issue by ueman** [fluttercommunity/plus_plugins#615]
## Use case

The `device_info_plus` plugin is often used to get more device information for crash reports or similar reasons.
In this use-case it's really cumbersome to use because you first have to detect the platform and then use the correct getter for the current device information. Therefor, it would be really useful if there's a method which gets the current device information.

## Proposal
I would like to have a method like the following example:
```dart
DeviceInfoPlugin deviceInfo = DeviceInfoPlugin();
// gets current device information
Map<String, dynamic> deviceInfo = await deviceInfo.info;
```

**Comment by RuzenDev** [comment]
Yes I agree on that too..

**Comment by Boehrsi** [comment]
I also like the idea to have one generic call to obtain the data. I would add the missing `toMap` calls for the platform specific objects (`LinuxDeviceInfo`, `WindowsDeviceInfo`) and would propose to add https://pub.dev/packages/universal_platform as dependency to create one info call which can deliver data without information regarding the platform.
I will prepare a PR.

**Comment by ueman** [comment]
> would propose to add https://pub.dev/packages/universal_platform as dependency

You don't need to do that. Each platform package needs to provide the generic method. The `device_info_plus` package (not the platform) can simply delegate to the platform package.

> I will prepare a PR.

Awesome, I very much appreciate it.

**Comment by Boehrsi** [comment]
> You don't need to do that. Each platform package needs to provide the generic method. The device_info_plus package (not the platform) can simply delegate to the platform package.

Thanks for the hint, this makes absolutely sense. 👍

**Comment by Boehrsi** [comment]
I think my PR isn't needed anymore as @alexbatalov PR already covers the required use case, right?

**Comment by ueman** [comment]
Yeah, I think so too. You could review it.

**Comment by Boehrsi** [comment]
Already had a brief look and it looks pretty good. If time allows I will have another in depth look.

**Comment by ueman** [comment]
This was implemented by #618
